### PR TITLE
PDCL-4583 For npm package, validate options and support an array-base…

### PR DESCRIPTION
**Pull request by Aaronius**
…d monitors option.

## Description
See https://jira.corp.adobe.com/browse/PDCL-4583 for more info.

It looks like our tests for the npm package are pretty sparse at the moment. I created a separate issue to address this: 
https://jira.corp.adobe.com/browse/PDCL-4584

I've manually tested it in the meantime.

## Related Issue
https://jira.corp.adobe.com/browse/PDCL-4583

## Motivation and Context
Simpler, more expected API. Better validation for a better user experience.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,14 +18,23 @@ import { createExecuteCommand } from "./core";
...
 const createNamespacedStorage = injectStorage(window);
-// set this up as a function so that monitors can be added at anytime
-// eslint-disable-next-line no-underscore-dangle
-const defaultGetMonitors = () => window.__alloyMonitors || [];
 
-export const createInstance = ({ name, getMonitors = defaultGetMonitors }) => {
+export const createInstance = options => {
+  const eventOptionsValidator = objectOf({
+    name: string().required(),
+    monitors: arrayOf(objectOf({}))
```
Comment (jonsnyder) [inline src/index.js:29]: you could do this:

monitors: arrayOf(objectOf({})).default([])

Even name could probably default to "alloy".

File: src/index.js [hunk 2]
```diff
@@ -18,14 +18,23 @@ import { createExecuteCommand } from "./core";
...
+  })
+    .noUnknownFields()
+    .required();
+  const { name, monitors = [] } = eventOptionsValidator(options);
+
+  // this is a function so that window.__alloyMonitors can be set or added to at any any time
```
Comment (jonsnyder) [inline src/index.js:35]: any any time

**Review by jonsnyder — APPROVED** [review]
Thanks for pushing for a better API before we release.

**Comment by jonsnyder** [comment]
I updated the GitHub wiki on how to use monitors with the NPM package. https://github.com/adobe/alloy/wiki/Monitoring-Hooks

**Comment by Aaronius** [comment]
Good suggestions. The changes have been made.

**Review by jfkhoury — APPROVED** [review]
Nice

**Pull request merged**